cdap-ui/app/cdap/services/WizardStores/MicroserviceUpload/ActionCreator.js

```diff
@@ -0,0 +1,155 @@
+  let url = `/namespaces/${namespace}/artifacts/${name}`;
+  let headers = {
+    'Content-Type': 'application/octet-stream',
+    'Artifact-Version': version,
+    'Artifact-Extends': state.uploadjson.artifactExtends
+  };
+  if (window.CDAP_CONFIG.securityEnabled) {
+    let token = cookie.load('CDAP_Auth_Token');
+    if (!isNil(token)) {
+      headers.Authorization = `Bearer ${token}`;
+    }
+  }
+  return UploadFile({url, fileContents: state.uploadjar.contents, headers});
+};
+
+const uploadConfigurationJson = () => {
+  const state = MicroserviceUploadStore.getState();
+  let {name:artifactId, version} = state.uploadjar.fileMetadataObj;
+  let namespace = NamespaceStore.getState().selectedNamespace;
+  let artifactConfigurationProperties = state.uploadjson.properties;
+  return MyArtifactApi
+    .loadPluginConfiguration({
+      namespace,
+      artifactId,
+      version
+    }, artifactConfigurationProperties);
+};
+
+const createApplication = () => {
+  const state = MicroserviceUploadStore.getState();
+  let namespace = NamespaceStore.getState().selectedNamespace;
+  let { instanceName: appId, description: appDescription, version: appVersion, microserviceName: pluginId } = state.general;
+  let { name: artifactId, version: artifactVersion } = state.uploadjar.fileMetadataObj;
+  let { instances, vcores, memory, ethreshold } = state.configure;
+
+  let config = {
+    version: appVersion,
+    id: appId,
+    description: appDescription,
+    plugin: {
+      name: pluginId,
+      artifact: {
+        name: artifactId,
+        version: artifactVersion,
+        scope: 'user'
+      }
+    },
+    configuration: {
+      instances,
+      vcores,
+      memory,
+      ethreshold
+    }
+  };
+
+  let endpoints = state.endpoints;
+  let endpointsObj = {
+    fetch: '',
+    in: [],
+    out: []
+  };
+
+  if (typeof(endpoints.fetch) !== 'number') {
+    delete endpointsObj.fetch;
+  } else {
+    endpointsObj.fetch = endpoints.fetch;
+  }
+
+  endpoints.in.forEach((inboundQueue) => {
+    if (inboundQueue.property.length > 0) {
+      endpointsObj.in.push(inboundQueue.property);
+    }
+  });
+  if (endpointsObj.in.length === 0) {
+    delete endpointsObj.in;
+  }
+
+  endpoints.out.forEach((outboundQueue) => {
+    if (outboundQueue.property.length > 0) {
+      endpointsObj.out.push(outboundQueue.property);
+    }
+  });
+  if (endpointsObj.out.length === 0) {
+    delete endpointsObj.out;
+  }
+
+  if (!isEmpty(endpointsObj)) {
+    config.configuration.endpoints = endpointsObj;
+  }
+
+  let properties = state.properties;
+  let propertiesKeyVal = properties.keyValues.pairs;
+  let propertiesObj = {};
+  propertiesKeyVal.forEach((pair) => {
+    if (pair.key.length > 0 && pair.value.length > 0) {
+      propertiesObj[pair.key] = pair.value;
+    }
+  });
+  if (!isEmpty(propertiesObj)) {
+    config.configuration.properties = propertiesObj;
+  }
+
+  let artifact = {
+    name: 'microservice-app',
+    version: '1.0-SNAPSHOT',
```

Comment:
even if you use a version range, you're hardcoding the scope on the next line, which will cause problems.